md/README.md

```diff
@@ -0,0 +1,121 @@
+
+This is a mention detection module that aims to annotate all mentions in given texts. It has an annotator that accepts a TextAnnotation and give it a new view which contains all the mentions as Constituents.
+
+This package tags all the tokens in a text into either "B/I/O" or "B/I/O/L/U" schema, then train/test on each single token. After predicting all tokens, we interpret this representation back to mentions.
+
+## Extent Detection
+
+The mentions predicted from the previous step is only mention heads. After detecting all mention-heads first, this package also has extent classification for each heads. In the extent classifier training process, each token in the extent to both right and left of a given mention head is formed into pairs to train a binary classifier, indicating whether the token is in the extent or not. To add negative examples, one token to the left and right of the mention extent is added. In the evaluating process, given a mention head, the package forms pairs with tokens to the left and right, until finding a token that is predicted to be not in the extent.
+
+## Results
+
+Head Boundary
+
+| Train\Eval (F1) | ACE  | ERE  |
+|-----------------|------|------|
+| ACE             | 89.6 | 83.8 |
+| ERE             | 81.7 | 86.7 |
+| ACE+ERE         | 88.4 | 86.5 |
+
+Extent boundary accuracy given Head
+
+ACE: 89.45
+
+## Using Annotator
+
+```java
+import edu.illinois.cs.cogcomp.annotation.AnnotatorException;
+import edu.illinois.cs.cogcomp.core.datastructures.textannotation.TextAnnotation;
+import edu.illinois.cs.cogcomp.core.datastructures.textannotation.Constituent;
+import edu.illinois.cs.cogcomp.core.datastructures.textannotation.View;
+import edu.illinois.cs.cogcomp.nlp.utility.TokenizerTextAnnotationBuilder;
+import edu.illinois.cs.cogcomp.annotation.TextAnnotationBuilder;
+import edu.illinois.cs.cogcomp.core.datastructures.ViewNames;
+import edu.illinois.cs.cogcomp.nlp.tokenizer.StatefulTokenizer;
+import org.cogcomp.md.MentionAnnotator;
+
+import java.io.IOException;
+import java.util.List;
+
+public class app
+{
+    public static void main( String[] args ) throws IOException, AnnotatorException
+    {
+        String text1 = "Good afternoon, gentlemen. I am a HAL-9000 "
+                + "computer. I was born in Urbana, Il. in 1992";
+
+        String corpus = "2001_ODYSSEY";
+        String textId = "001";
+
+        // Create a TextAnnotation using the LBJ sentence splitter
+        // and tokenizers.
+        TextAnnotationBuilder tab;
+        // don't split on hyphens, as NER models are trained this way
+        boolean splitOnHyphens = false;
+        tab = new TokenizerTextAnnotationBuilder(new StatefulTokenizer(splitOnHyphens));
+
+        TextAnnotation ta = tab.createTextAnnotation(corpus, textId, text1);
+
+        MentionAnnotator mentionAnnotator = new MentionAnnotator();
```

Comment:
So here you briefly have to explain how to configure this in order to load different models.